Source/JavaScriptCore/API/ObjCCallbackFunction.mm

```diff
@@ -727,13 +727,13 @@ JSObjectRef objCCallbackFunctionForMethod(JSContext *context, Class cls, Protoco
 {
     NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:[NSMethodSignature signatureWithObjCTypes:types]];
     [invocation setSelector:sel];
-    if (!isInstanceMethod) {
-        [invocation setTarget:cls];
-        // We need to retain the target Class because m_invocation doesn't retain it by default (and we don't want it to).
-        // FIXME: What releases it?
-        CFRetain((__bridge CFTypeRef)cls);
```

Comment:
> ensuring memory is not leaked, as nothing that releases the Class object could be found

Your patch does not change anything about the lifetime here; both retain and copy result in +1 retain count. And, I don't think preferring copy over retain makes any sense, nor does deleting the comment ("what releases it?") that is clearly still relevant.

Reply:
void destroy(Heap& heap) seems to release the memory too. Hopefully I am not wrong on this